test/common/config/new_grpc_mux_impl_test.cc

```diff
@@ -126,11 +184,17 @@ TEST_F(NewGrpcMuxImplTest, DynamicContextParameters) {
   // Update to bar type should resend Node.
   expectSendMessage("bar", {}, {});
   local_info_.context_provider_.update_cb_handler_.runCallbacks("bar");
-  expectSendMessage("foo", {}, {"x", "y"});
+  if (!isUnifiedMuxTest()) {
+    // in "legacy" delta mux implementation destruction of "foo_sub"
```

Comment:
Is this due to an issue in the legacy-delta implementation? Is there an issue# to add to this?

Reply:
Nods, In NewGrpcMuxImpl `addWatch` call returns a wrapper a `Watch`, which also has a dtor (here: https://github.com/envoyproxy/envoy/blob/main/source/common/config/new_grpc_mux_impl.h#L114) that's being called when `foo_sub` goes out of scope, and an unsubscribe message is sent. Unified mux implementation returns a thin wrapper around a `Watch` that's there for compatibility with existing interface only (So I don't need to introduce a very similar `addWatch` method). This compatibility wrapper doesn't have perform an unsubscribe on destruction, hence the difference.